lua/apisix/balancer.lua

```diff
@@ -47,18 +47,27 @@ end
 
 
 local function fetch_health_nodes(upstream, checker)
+    local up_nodes = core.table.new(0, #upstream.nodes)
+
     if not checker then
+        for addr, weight in pairs(upstream.nodes) do
+            if weight > 0 then
+                up_nodes[addr] = weight
+            end
+        end
         return upstream.nodes
```

Comment:
Here should `return up_nodes`

Reply:
old implementation is enough, it support weight equal 0 case already. we do not need to patch here, I reverted those new code.